### ISSUE TITLE
[Accessibility] Multiple issues with "Take action: N recommendations" buttons

### ISSUE THREAD

**Issue by PeterSaathoff-Harshfield** [ampproject/amp.dev#4505]
Environment:
Any OS/browser and screen reader (such as VoiceOver, NVDA, or ChromeVox)

Steps to reproduce:
1. Open https://amp-dev-staging.appspot.com/page-experience/
2. Enter a valid URL and press Analyze
3. Wait for the analysis to finish
4. Press tab repeatedly to move keyboard focus through controls
5. Attempt to move keyboard focus to the "Take action: N recommendations" button

Expected:
It should be possible to focus these buttons with the keyboard and activate them by pressing enter.

Actual:
These buttons are only useable with the mouse.

6. Turn on a screen reader
7. Navigate to one of the "Take action: N recommendations" buttons

Expected:
Screen readers should read the exact same text that is shown visually.

Actual:
Screen readers read these buttons as "Filter bylcp" or "Filter byfid button", for example.

8. Use a screen reader to activate the button

Expected:
Something happens that is understandable and meaningful to a screen reader user.

Actual:
The page scrolls to the "Take action. Improve your AMP site" section
Screen reader focus is not affected, so a screen reader user will think nothing happened or that something isn't working.

Once keyboard focusability is implemented on these buttons, keyboard focus should also do something meaningful when the button is pressed, rather than remaining on the button when the page scrolls.

The way I typically see this sort of "jump to heading" functionality implemented is by using a link with href="/#heading-name". In any case, this button should do something useful to and understandable by keyboard and screen reader users.

![image](https://user-images.githubusercontent.com/30604285/92973846-cd209800-f439-11ea-9ed2-b02232b4da22.png)

**Comment by caroqliu** [comment]
@PeterSaathoff-Harshfield To clarify, the "1 recommendation" `span` element does not need an `aria-label` because the text read by the focus should be the same as the text that is present in the tag?

**Comment by PeterSaathoff-Harshfield** [comment]
That's right, there should not be a label that differs from text on a button.
